add yoga tablet 3 8 support

exkc opened the pull request.

TravMurav commented:
This looks fine to me, but since the device was not ever upstream, could you do the following:

- create a revert commit for the original dts that is already in the tree
- create a single new commit adding two devices

You will end up with something similar to this: https://lore.kernel.org/linux-arm-msm/20240729-msm89xx-wingtech-init-v3-2-32c35476f098@trvn.ru/

I will also appreciate if you add a commit message to it similarly to the example ^^


(...) Then I will be able to remove original commit and the revert to replace it with your new commit and keep our history clean so it's easier to send upstream later

Commit:
Revert "ARM: dts: qcom: msm8909-lenovo-lxf-p5100: Add fuel gauge"

This reverts commit 0ae7357eff2ca7580505cfc5aa38334e678fa54f.

Signed-off-by: Karl Chan <exxxxkc@getgoogleoff.me>

Commit:
Revert "ARM: dts: qcom: msm8909-lenovo-lxf-p5100: Add initial device tree"

This reverts commit d71d3ce2ccf0974beb4d2336c07319885f46e864.

Signed-off-by: Karl Chan <exxxxkc@getgoogleoff.me>

Commit:
ARM: dts: qcom: qcom-msm8909-lenovo-lxf-p5x00: Add initial device tree for Lenovo p50000/p51000

Add initial device-tree for Lenovo Yoga Tab 3 8 LTE (lxf-p50000) and Lenovo Yoga Tab 3 10 LTE (lxf-p5100), which are MSM8909-based devices. These devices are quite
similar, so some configuration is shared in qcom-msm8909-lenovo-lxf-p5x00.dtsi.

Lenovo Yoga Tab 3 8 LTE (lxf-p50000):

- GPIO keys
- pm8909-vibrator
- SDHCI (internal and external storage)
- UART
- WCNSS (WiFi/BT)
- Regulators
- Touchscreen
- Fuel Gauge
- PWM panel backlight
- Hall
- Power LED
- Framebuffer
- USB (Dont Work)
- Magnetometer

Lenovo Yoga Tab 3 10 LTE (lxf-p5100):

- GPIO keys
- pm8909-vibrator
- SDHCI (internal and external storage)
- UART
- WCNSS (WiFi/BT)
- Regulators
- Touchscreen
- Fuel Gauge
- PWM panel backlight
- Hall
- Power LED
- Framebuffer
- USB (Dont Work)
- Magnetometer

Signed-off-by: Karl Chan <exxxxkc@getgoogleoff.me>

exkc commented:
done

exkc commented:
done

exkc commented:
oh wait i mady a tyo let me fix it

exkc commented:
alr done

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5x00.dtsi:
```diff
@@ -0,0 +1,380 @@
+	status = "okay";
+
+	fuel-gauge@55 {
+		monitored-battery = <&battery>;
+		compatible = "ti,bq27500";
+		reg = <0x55>;
```
One small nit:

```suggestion
		compatible = "ti,bq27500";
		reg = <0x55>;
		monitored-battery = <&battery>;
```

https://elixir.bootlin.com/linux/v6.14-rc4/source/Documentation/devicetree/bindings/dts-coding-style.rst#L117-L124

exkc commented:
done

TravMurav commented:
Manually applied https://github.com/msm8916-mainline/linux/commit/d71adc8c65f42d46aeb5c92173c639447e563b54 to the new wip/msm8916/6.15 branch, thanks and sorry for the delay
